java-dynamic-sqs-listener-core/src/main/java/com/jashmore/sqs/util/Immutables.java

```diff
@@ -0,0 +1,66 @@
+package com.jashmore.sqs.util;
+
+import static java.util.stream.Collectors.collectingAndThen;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toSet;
+
+import lombok.experimental.UtilityClass;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Utility class used to provide helper methods to construct Immutable implementations of data structures.
+ *
+ * <p>This has been implemented again to reduce dependencies on other modules, more specifically Guava.
+ */
+@UtilityClass
```

Comment:
Interesting... did not know about this

Reply:
Yeah not incredibly useful but does make sure you can't construct it.